velox/exec/MergeJoin.cpp

```diff
@@ -273,11 +299,20 @@ void MergeJoin::prepareOutput() {
     outputSize_ = 0;
 
     if (filterInput_ != nullptr) {
-      // When filterInput_ contains array or map columns, their child vectors
-      // (elements, keys and values) keep growing after each call to
-      // 'copyRow'. Call BaseVector::resize(0) on these child vectors to avoid
-      // that.
-      for (auto& child : filterInput_->children()) {
+      for (auto i = 0; i < filterInputType_->size(); ++i) {
+        auto& child = filterInput_->childAt(i);
+        // If 'child' is also projected to output, make 'child' to use the same
+        // shared pointer to the output column.
+        if (filterInputToOutputChannel_.find(i) !=
+            filterInputToOutputChannel_.end()) {
+          child = output_->childAt(filterInputToOutputChannel_[i]);
+          continue;
+        }
+
+        // When filterInput_ contains array or map columns that are not
+        // projected to output, their child vectors(elements, keys and values)
+        // keep growing after each call to 'copyRow'. Call BaseVector::resize(0)
```

Comment:
> Call BaseVector::resize(0)

It is confusing that the comment says "Call resize(0)", but the code does something else.